fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/ItemStackMixin.java

```diff
@@ -48,33 +48,23 @@
 public abstract class ItemStackMixin implements FabricItemStack {
 	@Shadow public abstract Item getItem();
 
-	@Unique
-	private LivingEntity fabric_damagingEntity;
-
-	@Unique
-	private Consumer<LivingEntity> fabric_breakCallback;
-
-	@Inject(method = "damage(ILnet/minecraft/entity/LivingEntity;Ljava/util/function/Consumer;)V", at = @At("HEAD"))
-	private void saveDamager(int amount, LivingEntity entity, Consumer<LivingEntity> breakCallback, CallbackInfo ci) {
-		this.fabric_damagingEntity = entity;
-		this.fabric_breakCallback = breakCallback;
-	}
-
-	@ModifyArg(method = "damage(ILnet/minecraft/entity/LivingEntity;Ljava/util/function/Consumer;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;damage(ILnet/minecraft/util/math/random/Random;Lnet/minecraft/server/network/ServerPlayerEntity;)Z"), index = 0)
-	private int hookDamage(int amount) {
+	@WrapOperation(method = "damage(ILnet/minecraft/entity/LivingEntity;Lnet/minecraft/entity/EquipmentSlot;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;damage(ILnet/minecraft/util/math/random/Random;Lnet/minecraft/server/network/ServerPlayerEntity;Ljava/lang/Runnable;)V"))
+	private void hookDamage(ItemStack instance, int amount, Random random, ServerPlayerEntity serverPlayerEntity, Runnable runnable, Operation<Void> original, @Local(argsOnly = true) EquipmentSlot slot) {
 		CustomDamageHandler handler = ((ItemExtensions) getItem()).fabric_getCustomDamageHandler();
 
 		if (handler != null) {
-			return handler.damage((ItemStack) (Object) this, amount, fabric_damagingEntity, fabric_breakCallback);
+			// Track whether an item has been broken by custom handler
+			MutableBoolean mut = new MutableBoolean(false);
+			amount = handler.damage((ItemStack) (Object) this, amount, serverPlayerEntity, slot, () -> {
+				mut.setTrue();
+				runnable.run();
+			});
+
+			// If item is broken, there's no reason to call the original.
+			if (mut.booleanValue()) return;
 		}
```

Comment:
Since we're using a boolean anyways, wouldn't `WrapWithCondition` be easier to use here?